pkg/fuzzy/aws_translate.go

```diff
@@ -0,0 +1,173 @@
+	awst := &AWSTranslate{}
+
+	for _, opt := range opts {
+		if optErr := opt(awst); optErr != nil {
+			return nil, fmt.Errorf("apply opt: %w", optErr)
+		}
+	}
+
+	// Ping the AWS Translate API to ensure that the client is working.
+	_, err := awst.client.TranslateText(ctx, &translate.TranslateTextInput{
+		SourceLanguageCode: ptr("en"),
+		TargetLanguageCode: ptr("lv"),
+		Text:               ptr("Hello World!"),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("AWS translate client: ping AWS translate: %w", err)
+	}
+
+	return awst, nil
+}
+
+// --------------------Methods--------------------
+
+func (a *AWSTranslate) Translate(ctx context.Context, messages *model.Messages) (*model.Messages, error) {
+	if messages == nil {
+		return nil, nil
+	}
+
+	if len(messages.Messages) == 0 {
+		return &model.Messages{Language: messages.Language, Original: messages.Original}, nil
+	}
+
+	// skip locale part if region is not a country,
+	// AWS only supports ISO 3166 2-digit country codes.
+	// https://docs.aws.amazon.com/translate/latest/dg/what-is-languages.html
+	targetLanguage := messages.Language.String()
+
+	if region, _ := messages.Language.Region(); !region.IsCountry() {
+		baseLang, _ := messages.Language.Base()
+		targetLanguage = baseLang.String()
+	}
+
+	translatedMessages := model.Messages{
+		Language: messages.Language,
+		Original: messages.Original,
+		Messages: make([]model.Message, 0, len(messages.Messages)),
+	}
+
+	for _, m := range messages.Messages {
+		translateOutput, err := a.client.TranslateText(ctx,
+			&translate.TranslateTextInput{
+				// Amazon Translate supports text translation between the languages listed in the following table.
+				// The language code column uses ISO 639-1 two-digit language codes.
+				// For a country variant of a language, the table follows the RFC 5646 format of appending a dash
+				// followed by an ISO 3166 2-digit country code.
+				// For example, the language code for the Mexican variant of Spanish is es-MX.
+				// List of supported languages - https://docs.aws.amazon.com/translate/latest/dg/what-is-languages.html
+				TargetLanguageCode: ptr(targetLanguage),
+
+				// TODO: replace auto detect with messages source language.
+				// If you specify auto , you must send the TranslateText request in a region that supports Amazon Comprehend.
+				SourceLanguageCode: ptr("auto"),
+				// Maximum text size limit accepted by the AWS Translate API - 10000 bytes.
+				Text: ptr(m.Message),
+			})
+		if err != nil {
+			return nil, fmt.Errorf("AWS translate text, message id '%s': %w", m.ID, err)
+		}
+
+		translatedMessages.Messages = append(translatedMessages.Messages, model.Message{
+			ID:          m.ID,
+			PluralID:    m.PluralID,
+			Description: m.Description,
+			Message:     *translateOutput.TranslatedText,
+			Status:      model.MessageStatusFuzzy,
+			Positions:   m.Positions,
+		},
+		)
+	}
+
+	if len(translatedMessages.Messages) != len(messages.Messages) {
```

Comment:
Is this common, that the texts slice that you receive is longer than the input? If not that this is redundant

Reply:
Janis Kļaviņš wrote this code and I think it might happen in theory, so I decided to leave these lines of code.

Reply:
Please remove it, it should never happen. @klavinsjanis overengineered it initially.